File: python/a32.py
# A32 Game 1
# submission: https://atcoder.jp/contests/tessoku-book/submissions/38677956

n, a, b = map(int, input().split())

win = [False] * (n + 1)

for i in range(n + 1):
    if win[i] == False:
        if i + a <= n:
            win[i + a] = True
        if i + b <= n:
            win[i + b] = True

if win[n]:
    print('First')
else:
    print('Second')

File: rust/tessoku-book/src/bin/a32.rs
// A32 Game 1
// submission: https://atcoder.jp/contests/tessoku-book/submissions/38678056

use proconio::input;

fn main() {
    input!{
        n: usize,
        a: usize,
        b: usize,
    }
    let mut win = vec![false; n + 1];
    for i in 0..=n {
        if win[i] == false {
            if i + a <= n {
                win[i + a] = true;
            }
            if i + b <= n {
                win[i + b] = true;
            }
        }
    }
    if win[n] {
        println!("First");
    } else {
        println!("Second");
    }
}
